src/AppInstallerCLITests/YamlManifest.cpp

```diff
@@ -886,6 +888,198 @@ TEST_CASE("ValidateV1_6GoodManifestAndVerifyContents", "[ManifestValidation]")
     VerifyV1ManifestContent(mergedManifest, false, ManifestVer{ s_ManifestVersionV1_6 });
 }
 
+TEST_CASE("WriteV1SingletonManifestAndVerifyContents", "[ManifestCreation]")
+{
+    ManifestValidateOption validateOption;
+    TempDirectory singletonDirectory{ "SingletonManifest" };
+    CopyTestDataFilesToFolder({ "ManifestV1_1-Singleton.yaml" }, singletonDirectory);
+    Manifest singletonManifest = YamlParser::CreateFromPath(singletonDirectory, validateOption);
+
+    TempDirectory exportedSingletonDirectory{ "exportedSingleton" };
+    std::filesystem::path generatedSingletonManifestPath = exportedSingletonDirectory.GetPath() / "testSingletonManifest.yaml";
+    YamlWriter::OutputYamlFile(singletonManifest, singletonManifest.Installers[0], generatedSingletonManifestPath);
+
+    REQUIRE(std::filesystem::exists(generatedSingletonManifestPath));
+    Manifest generatedSingletonManifest = YamlParser::CreateFromPath(exportedSingletonDirectory, validateOption);
+    VerifyV1ManifestContent(generatedSingletonManifest, true);
+
+    TempDirectory multiFileDirectory{ "MultiFileManifest" };
+    CopyTestDataFilesToFolder({
+        "ManifestV1-MultiFile-Version.yaml",
+        "ManifestV1-MultiFile-Installer.yaml",
+        "ManifestV1-MultiFile-DefaultLocale.yaml",
+        "ManifestV1-MultiFile-Locale.yaml" }, multiFileDirectory);
+
+    Manifest multiFileManifest = YamlParser::CreateFromPath(multiFileDirectory, validateOption);
+    TempDirectory exportedMultiFileDirectory{ "exportedMultiFile" };
+    std::filesystem::path generatedMultiFileManifestPath = exportedMultiFileDirectory.GetPath() / "testMultiFileManifest.yaml";
+    YamlWriter::OutputYamlFile(multiFileManifest, multiFileManifest.Installers[0], generatedMultiFileManifestPath);
+
+    REQUIRE(std::filesystem::exists(generatedMultiFileManifestPath));
+    Manifest generatedMultiFileManifest = YamlParser::CreateFromPath(exportedMultiFileDirectory, validateOption);
+    VerifyV1ManifestContent(generatedMultiFileManifest, true);

```

Comment:
>VerifyV1ManifestContent [](http://example.com/codeflow?start=4&length=23)

The isSingleton should be false and additional locales should be verified. Regarding installer, we could have another bool called isExport then skip defaultInstallerInfo and additional installer if needed.

Ideally, the manifest export code should support exporting a single installer and exporting all installers if no specific installer is passed in. But I'm ok not doing exporting all installers case in this change since it's not used anywhere yet.

Reply:
Added a TODO comment for outputting all installers if no single installer is present. 

Updated tests to behave differently if the manifest is exported.